### PR TITLE
Bring profile documentation up to date.

### PR DESCRIPTION
**Pull request by meisterT**
Precursor to addressing https://github.com/bazelbuild/bazel/issues/11384

RELNOTES: None

### CODE REVIEW

File: site/docs/skylark/performance.md [hunk 1]
```diff
@@ -194,13 +194,113 @@ ctx.actions.run(
...
-To profile your code and analyze the performance you have two options:
+Bazel writes a JSON profile by default to the file `command.profile.gz` in the
+output base. The location can be configured using the `--profile` flag, for
+example `--profile=/tmp/profile.gz`. If that location ends with `.gz`, the
+profile will be compressed with GZIP.
 
-- use the `--profile` flag and the `analyze-profile` command, or
-- use the new `--experimental_generate_json_trace_profile` flag and load the
-  resulting JSON profile in `chrome://tracing` (recommended).
+The resulting profile (potentially compressed) profile file can be loaded and
```
Comment (joeleba) [inline site/docs/skylark/performance.md:202]: "The resulting ~profile~ (potentially compressed) profile file"

File: site/docs/skylark/performance.md [hunk 2]
```diff
@@ -194,13 +194,113 @@ ctx.actions.run(
...
+*   Slower than expected analysis phase (`runAnalysisPhase`), especially on
+    incremental builds. This can be a sign of a poor rule implementation, for
+    example one that flattens depsets. Package loading can be slow by an
+    excessive amount of targets, complex macros or recursive globs.
+*   Individual slow actions, especially those on the critical path. It might be
+    possible to split large actions into multiple smaller actions or reduce the
+    set of (transitive) dependencies to speed them up. Also check for an unusual
+    high non-`PROCESS_TIME` (e.g. `REMOTE_SETUP` or `FETCH`).
+*   Bottlenecks, that is a small number of threads is busy while all others are
+    idleing waiting for the result (see around 15s-30s in above screenshot).
```
Comment (joeleba) [inline site/docs/skylark/performance.md:256]: s/idleing/idling

File: site/docs/skylark/performance.md [hunk 3]
```diff
@@ -194,13 +194,113 @@ ctx.actions.run(
...
+    Optimizing this will most likely require touching the rule implementations
+    or Bazel changes to introduce more parallelism. This can also happen when
```
Comment (joeleba) [inline site/docs/skylark/performance.md:258]: s/changes/itself

**Comment by meisterT** [comment]
cc @floriographygoth

**Pull request closed without merging**